components/authz-service/storage/v2/postgres/postgres.go

```diff
@@ -865,10 +866,31 @@ func (p *pg) CreateProject(ctx context.Context, project *v2.Project) (*v2.Projec
 	ctx, cancel := context.WithCancel(ctx)
 	defer cancel()
 
-	if err := p.insertProjectWithQuerier(ctx, project, p.db); err != nil {
+	tx, err := p.db.BeginTx(ctx, nil)
+	if err != nil {
+		return nil, p.processError(err)
+	}
+
+	row := tx.QueryRowContext(ctx, ` WITH t as (SELECT id from iam_projects WHERE type='custom') SELECT COUNT(*) FROM t;`)
+	var numProjects int64
+	if err := row.Scan(&numProjects); err != nil {
+		return nil, p.processError(err)
+	}
+
+	p.logger.Warnf("projects %d", numProjects)
+	if numProjects >= v2_constants.MaxProjects {
```

Comment:
We're in trouble if this is already exceeded . . .

Reply:
yeah i saw got an error when i added 6 projects on v1, then tried to upgrade to v2.
```
Upgrading to IAM v2
Migrating v1 policies...
UnknownError: Failed to upgrade IAM to v2: Failed to reset IAM v2 database state: rpc error: code = Internal desc = reset to default project: max projects allowed
```
which comes from [this line](https://github.com/chef/automate/blob/master/components/authz-service/server/v2/policy.go#L541)